storage/bootstrap/bootstrapper/peers/source.go

```diff
@@ -146,39 +157,8 @@ func (s *peersSource) Read(
 		xlog.NewField("incremental", incremental),
 	).Infof("peers bootstrapper bootstrapping shards for ranges")
 	if incremental {
-		// If performing an incremental bootstrap then flush one
-		// at a time as shard results are gathered
-		incrementalWg.Add(1)
-		go func() {
-			defer incrementalWg.Done()
-
-			for flush := range incrementalQueue {
-				err := s.incrementalFlush(persistFlush, flush.nsMetadata, flush.shard,
-					flush.shardRetrieverMgr, flush.shardResult, flush.timeRange)
-				if err == nil {
-					continue
-				}
-
-				// Remove results and make unfulfilled if an error occurred
-				s.log.WithFields(
-					xlog.NewField("error", err.Error()),
-				).Errorf("peers bootstrapper incremental flush encountered error")
-
-				// Remove results
-				tr := flush.timeRange
-				blockSize := nsMetadata.Options().RetentionOptions().BlockSize()
-				for _, series := range flush.shardResult.AllSeries() {
-					for at := tr.Start; at.Before(tr.End); at = at.Add(blockSize) {
-						series.Blocks.RemoveBlockAt(at)
-					}
-				}
-
-				// Make unfulfilled
-				lock.Lock()
-				result.Add(flush.shard, nil, xtime.Ranges{}.AddRange(tr))
-				lock.Unlock()
-			}
-		}()
+		go s.startIncrementalQueueWorker(
+			incrementalWorkerDoneCh, incrementalQueue, persistFlush, result, &lock)
```

Comment:
Perhaps to keep the flow readable from this function, do the closing of the worker channel inline here?

Reply:
I'm cool with the rename, but not sure what you're saying about the doneCh.

If I move this line:

```
<- incrementalWorkerDoneCh
```

into the goroutine like you have it, then the code won't wait for the incremental queue worker loop to complete before continuing.